### PR TITLE
ENH: dtype dispatcher for Dataframe Intercahnge Protocol in Python

### PR DESCRIPTION
**Pull request by samir-nasibli**
# Description
dtype dispatcher for Dataframe Intercahnge Protocol in Python
Duplicate from https://github.com/oneapi-src/oneDAL/pull/2555
Since I have some problems with the original fork, moved changes to this branch.

### CODE REVIEW

File: cpp/oneapi/dal/detail/dtype_dispatcher.hpp [hunk 1]
```diff
@@ -0,0 +1,128 @@
+/*******************************************************************************
+* Copyright 2024 Intel Corporation
```
Comment (Alexandr-Solovev) [inline cpp/oneapi/dal/detail/dtype_dispatcher.hpp:2]: Please use the updated copyright rules here

**Review by ethanglaser — APPROVED** [review]
Is there any functional change or just for reorganization?

**Comment by samir-nasibli** [comment]
> Is there any functional change or just for reorganization?

It is refactoring required on future python side Dataframe intercahnge protocol integration.

**Pull request merged**